xtraplatform-feature-provider-sql/src/main/java/de/ii/xtraplatform/feature/provider/sql/infra/db/SqlTypeInfoValidator.java

```diff
@@ -89,12 +95,45 @@ public SqlTypeInfoValidator(ConnectionInfoSql connectionInfo, ClassLoader classL
 
             relations.forEach(relation -> {
                 //TODO: tableExists for relation.getSourceContainer(), relation.getTargetContainer() and relation.getJunction() (if present)
+                if (!tableExists(relation.getSourceContainer())) {
+                    errors.add(String.format("table '%s' does not exist", relation.getSourceContainer()));
+                }
+
+                if (!tableExists(relation.getTargetContainer())) {
+                    errors.add(String.format("table '%s' does not exist", relation.getTargetContainer()));
+                }
 
                 //TODO: columnExists for relation.getSourceField(), relation.getSourceSortKey() and relation.getTargetField()
+                if (relation.getJunction().isPresent() && !tableExists(relation.getJunction().get())) {
+                    errors.add(String.format("table '%s' does not exist", relation.getJunction().get()));
+                }
+
+                if (!columnExists(relation.getSourceField(), relation.getSourceSortKey())) {
+                    errors.add(String.format("column '%s' in table '%s' does not exist", relation.getSourceField(), relation.getSourceSortKey()));
+                }
+
+                if (!columnExists(relation.getTargetField(), relation.getSourceSortKey())) {
+                    errors.add(String.format("column '%s' in table '%s' does not exist", relation.getTargetField(), relation.getSourceSortKey()));
+                }
 
                 //TODO: columnExists for relation.getJunctionSource() and relation.getJunctionTarget() if present
+                if (relation.getJunctionSource().isPresent() && relation.getJunctionTarget().isPresent()) {
+                    if (!columnExists(relation.getJunctionSource().get(), relation.getJunctionTarget().get())) {
+                        errors.add(String.format("column '%s' in table '%s' does not exist",
+                                relation.getJunctionSource().get(), relation.getJunctionTarget().get()));
+                    }
+                }
 
                 //TODO: isColumnUnique for relation.getSourceSortKey()
+                if (!isColumnUnique(relation.getSourceField(), relation.getSourceSortKey())) {
```

Comment:
The first parameter should be `relation.getSourceSortKey()`, the second `relation.getSourceContainer()`.